cmd/draino/draino.go

```diff
@@ -137,6 +139,22 @@ func main() {
 		}
 	}
 
+	if *noDrain {
```

Comment:
Instead of replicating the configuration of our handler on line 120, we could simply pass `*noDrain` into the `NewAPICordonDrainer` invocation e.g. `kubernetes.WithDrain(!*noDrain)`.

That looks a bit funky though. Rename the flag to simply `drain` and you won't have to invert it there. Kingping also automatically provides [inverted forms](https://github.com/alecthomas/kingpin#boolean-values) for boolean flags e.g. you'll get `--no-drain` and `--drain`. To maintain compatibility, set the default `Default(true)`.

Reply:
I've implemented this, but found that until kingpin v3, the `--no-drain` flag doesn't appear in the help (see https://github.com/alecthomas/kingpin/issues/216)

I've updated the README to reflect this, but wondering if you'd like to:

- Bump the deps to kingpin.v3-unstable
- Use the original logic so the `--no-drain` is clear without updating the deps
- Leave things as they are?

Reply:
Ah that's silly - did not realize this was a v3 thing and the project hasn't made a stable v3 release AFAICT. I'm fine with `--no-drain` in that case. Thanks for taking the time to humor this suggestion.